QuoteEngine.py

```diff
@@ -0,0 +1,30 @@
+import requests
+import yaml
+import random
+
+
+class QuoteEngine:
+
+    def __init__(self, logger):
+        self.quote_url = "https://cdn.rawgit.com/tulul/tulul-quotes/master/quote.yaml"  # noqa
+        # Note: rawgit does not have 100% uptime, but at
+        # least they're not throttling us.
+
+        self.logger = logger
+        self.cache = []
+        self.default_empty_error = "Nothing in cache. Forget to refresh_cache?"
+
+    def retrieve_random(self):
+        cache = self.cache
+        index = random.randrange(0, len(cache))
+        return self.format_quote(cache[index])
+
+    def format_quote(self, q):
+        return "%s - %s, %s" % (q['quote'], q['author'], q['author_bio'])
```

Comment:
The new way of formatting string is using `.format` method. This can be simplified like this:

```
return '{q[quote]} - {q[author]}, {q[author_bio]}'.format(q=q)
# or
return '{0[quote]} - {0[author]}, {0[author_bio]}'.format(q)
```